Support void return type in TensorIteratorDynamicCasting checks.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38817 Add dynamic_cast asserts to CPU Loops.
* **#38815 Support void return type in TensorIteratorDynamicCasting checks.**
* #38813 Move needs_dynamic_casting to a non-CUDA specific file.
* #38810 Improve error checking of CUDALoops.
* #38809 Improve cpu/Loops.h arity asserts.

Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Differential Revision: [D21670717](https://our.internmc.facebook.com/intern/diff/D21670717)

gchanan committed:
Support void return type in TensorIteratorDynamicCasting checks.

Some CPU kernels have void return types and the currently implementation segfaults on these cases.

[ghstack-poisoned]

zasdfgbnm commented:
Build errors seems to be caused by missing an include `#include <c10/core/ScalarType.h>`

gchanan committed:
Update on "Support void return type in TensorIteratorDynamicCasting checks."


Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Differential Revision: [D21670717](https://our.internmc.facebook.com/intern/diff/D21670717)

[ghstack-poisoned]

gchanan committed:
Update on "Support void return type in TensorIteratorDynamicCasting checks."


Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Differential Revision: [D21670717](https://our.internmc.facebook.com/intern/diff/D21670717)

[ghstack-poisoned]

gchanan committed:
Update on "Support void return type in TensorIteratorDynamicCasting checks."


Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Differential Revision: [D21670717](https://our.internmc.facebook.com/intern/diff/D21670717)

[ghstack-poisoned]

gchanan committed:
Update on "Support void return type in TensorIteratorDynamicCasting checks."


Some CPU kernels have void return types and the currently implementation segfaults on these cases.

Differential Revision: [D21670717](https://our.internmc.facebook.com/intern/diff/D21670717)

[ghstack-poisoned]
